lti_consumer/models.py

```diff
@@ -526,3 +528,58 @@ def __str__(self):
 
     class Meta:
         app_label = 'lti_consumer'
+
+
+class CourseEditLTIFieldsEnabledFlag(ConfigurationModel):
+    """
+    Enables the editing of "request username" and "request email" fields
+    of LTI consumer for a specific course.
+
+    .. no_pii:
+    """
+    KEY_FIELDS = ('course_id',)
+
+    course_id = CourseKeyField(max_length=255, db_index=True)
+
+    @classmethod
+    @request_cached
+    def lti_access_to_learners_editable(cls, course_id, is_already_sharing_learner_info):
+        """
+        Looks at the currently active configuration model to determine whether
+        the feature that enables editing of "request username" and "request email"
+        fields of LTI consumer is available or not.
+
+        Backwards Compatibility:
+        Enable this feature for a course run who was sharing learner username/email
+        in the past.
+
+        Arguments:
+            course_id (CourseKey): course id for which we need to check this configuration
+            is_already_sharing_learner_info (bool): indicates whether LTI consumer is
+            already sharing edX learner username/email.
+        """
+        course_specific_config = (CourseEditLTIFieldsEnabledFlag.objects
+                                  .filter(course_id=course_id)
+                                  .order_by('-change_date')
+                                  .first())
+
+        if is_already_sharing_learner_info:
+            if not course_specific_config:
+                CourseEditLTIFieldsEnabledFlag.objects.create(course_id=course_id, enabled=True)
+                return True
+
+        return course_specific_config.enabled if course_specific_config else False
+
+    def __str__(self):
+        en = "Not "
+        if self.enabled:
+            en = ""
+
+        return "Course '{course_id}': Edit LTI access to Learner information {en}Enabled".format(
+            course_id=str(self.course_id),
+            en=en,
+        )
+
+    class Meta:
+        # This model is moved from
+        db_table = "xblock_config_courseeditltifieldsenabledflag"
```

Comment:
Can add a little detail about why defining `db_table` here?

Reply:
Sorry I did add something but it seems at some point I accidentally removed it.

Reply:
Sorry forgot to push the latest commit.